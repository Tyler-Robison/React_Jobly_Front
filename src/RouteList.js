import React from "react";
import { Route, Routes, Navigate } from "react-router-dom";
import Home from "./Home";
import CompanyDetails from "./CompanyDetails";
import CompanyList from "./CompanyList";
import JobList from "./JobList";
import ProfileForm from "./ProfileForm";
import LoginForm from "./LoginForm";
import SignupForm from "./SignupForm";
import RequireAuth from "./RequireAuth";

const RouteList = ({
    companies,
    setCompanies,
    jobs,
    setJobs,
    login,
    displayMsg,
    clearMsg }) => {


    return (
        <Routes>
            <Route path="/jobly" element={<Home />} />


            {/* <Route path="/companies"
                element={<CompanyList companies={companies} setCompanies={setCompanies} />} /> */}

            <Route path="/companies"
                element={<RequireAuth redirectTo="/login"><CompanyList companies={companies} setCompanies={setCompanies} /></RequireAuth>} />


            <Route path="/companies/:handle"
                element={<RequireAuth redirectTo="/login"><CompanyDetails /></RequireAuth>} />


            {/* <Route path="/jobs"
                element={<JobList jobs={jobs} setJobs={setJobs} />} /> */}

            <Route path="/jobs"
                element={<RequireAuth redirectTo="/login"><JobList jobs={jobs} setJobs={setJobs} /></RequireAuth>} />


            {/* <Route path="/profile"
                element={<ProfileForm displayMsg={displayMsg} clearMsg={clearMsg} />} /> */}

            <Route path="/profile"
                element={<RequireAuth redirectTo="/login"><ProfileForm displayMsg={displayMsg} clearMsg={clearMsg} /></RequireAuth>} />


            <Route path="/signup" element={<SignupForm login={login} displayMsg={displayMsg} clearMsg={clearMsg} />} />
            <Route path="/login" element={<LoginForm login={login} displayMsg={displayMsg} clearMsg={clearMsg} />} />
            <Route path="/" element={<Navigate replace to="/jobly" />} />
            <Route path='*' element={<Navigate replace to="/jobly" />} />
        </Routes>
    )
}

export default RouteList;