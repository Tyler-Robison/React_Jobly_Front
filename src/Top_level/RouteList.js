import React from "react";
import { Route, Routes, Navigate } from "react-router-dom";
import Home from "../Home/Home";
import CompanyDetails from "../Company/CompanyDetails";
import CompanyList from "../Company/CompanyList";
import JobList from "../Job/JobList";
import ProfileForm from "../Forms/ProfileForm";
import LoginForm from "../Forms/LoginForm";
import SignupForm from "../Forms/SignupForm";
import RequireAuth from "./RequireAuth";

const RouteList = ({
    companies,
    setCompanies,
    jobs,
    setJobs,
    login,
    displayMsg,
    clearMsg,
    setCurrentUser }) => {


    return (
        <Routes>
            <Route path="/jobly" element={<Home />} />

            <Route path="/companies"
                element={<RequireAuth redirectTo="/login"><CompanyList companies={companies} setCompanies={setCompanies} /></RequireAuth>} />


            <Route path="/companies/:handle"
                element={<RequireAuth redirectTo="/login"><CompanyDetails /></RequireAuth>} />


            <Route path="/jobs"
                element={<RequireAuth redirectTo="/login"><JobList jobs={jobs} setJobs={setJobs} /></RequireAuth>} />


            <Route path="/profile"
                element={<RequireAuth redirectTo="/login"><ProfileForm displayMsg={displayMsg} setCurrentUser={setCurrentUser}/></RequireAuth>} />

            <Route path="/signup" element={<SignupForm login={login} displayMsg={displayMsg} clearMsg={clearMsg} />} />
            <Route path="/login" element={<LoginForm login={login} displayMsg={displayMsg} clearMsg={clearMsg} />} />
            <Route path="/" element={<Navigate replace to="/jobly" />} />
            <Route path='*' element={<Navigate replace to="/jobly" />} />
        </Routes>
    )
}

export default RouteList;