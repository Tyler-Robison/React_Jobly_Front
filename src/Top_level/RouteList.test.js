import { render, screen } from '@testing-library/react';
import RouteList from './RouteList';
import { MemoryRouter } from "react-router-dom";
import UserProvider from '../testUtils'


it("renders without crashing", function () {
    <MemoryRouter>
        <RouteList />
    </MemoryRouter>
});

it("matches snapshot", function () {

    const container = render(
        <MemoryRouter>
            <UserProvider>
                <RouteList />
            </UserProvider>
        </MemoryRouter>
    );
    expect(container.asFragment()).toMatchSnapshot();
});