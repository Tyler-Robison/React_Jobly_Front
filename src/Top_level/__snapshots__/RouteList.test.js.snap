// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <h1>
    Jobly
  </h1>
  <h3>
    All the jobs in one convenient place
  </h3>
  <h2>
    Welcome Back, Janet
  </h2>
</DocumentFragment>
`;
